components/Promotion/basic_usage.rst

```diff
@@ -0,0 +1,148 @@
+
+    $instruction = new Instruction(); // $amount = 5 by default
+
+    /**
+     * @param RepositoryInterface    $repository
+     * @param EntityManagerInterface $manager
+     */
+    $generator = new CouponGenerator($repository, $manager);
+
+    //This will generate and persist 5 coupons into the database
+    //basing on the instruction provided for the given promotion object
+    $generator->generate($promotion, $instruction);
+
+    // We can also generate one unique code, and assign it to a new Coupon.
+    $code = $gen->generateUniqueCode();
+    $coupon = new Coupon();
+    $coupon->setCode($code);
+
+
+.. _component_promotion_checker_promotion-eligibility-checker:
+
+PromotionEligibilityChecker
+---------------------------
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Core\Model\OrderItem;
+    use Sylius\Component\Promotion\Model\Rule;
+    use Sylius\Component\Promotion\Model\Action;
+    use Sylius\Component\Promotion\Checker\PromotionEligibilityChecker;
+
+    /**
+     * @param ServiceRegistryInterface $registry
+     * @param EventDispatcherInterface $dispatcher
+     */
+    $checker = new PromotionEligibilityChecker($registry, $dispatcher);
+
+    // Let's create a new promotion
+    $promotion = new Promotion();
+    $promotion->setName('Test');
+    $promotion->setUsageLimit(0);
+    $promotion->setUsed(0);
+
+    // And a new action for that promotion, that will give a fixed discount of 10
+    $action = new Action();
+    $action->setType('fixed_discount');
+    $action->setConfiguration(array('amount' => 10));
+    $action->setPromotion($promotion);
+
+    // That promotion will also have a rule - works for item amounts over 3
+    $rule = new Rule();
+    $rule->setType('item_count');
+    $rule->setConfiguration(array('count' => 3));
+
+    $promotion->addRule($rule);
+
+    // Now we need an object that implements the PromotionSubjectInterface
+    $orderItem = new OrderItem();
+    $orderItem->setUnitPrice(10);
+    $orderItem->setQuantity(4);
+    $orderItem->addPromotion($promotion);
+
+    $checker->isEligible($orderItem, $promotion); // Returns true
+
+.. note::
+
+    It implements ``PromotionEligibilityCheckerInterface``
+
+
+.. _component_promotion_action_promotion-applicator:
+
+PromotionApplicator
+-------------------
+
+In order to automate the process of promotion application the component provides us with a Promotion Applicator.
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Action\PromotionApplicator;
+    use Sylius\Component\Registry\ServiceRegistry;
+    use Sylius\Component\Promotion\Model\Promotion;
+
+    $registry = new ServiceRegistry('interface');
+    $promotionApplicator = new PromotionApplicator($registry);
+
+    $promotion = new Promotion();
+
+    /**
+     * @var PromotionSubjectInterface $subject
+     */
+    $subject = /* ... */
+    $subject->addPromotion($promotion);
```

Comment:
Ah no, wait... Actually, you are missing the second argument, see here - https://github.com/Sylius/Promotion/blob/master/Action/PromotionApplicator.php#L32 / http://api.sylius.org/Sylius/Component/Promotion/Action/PromotionApplicator.html#method_apply